connectors/sources/tests/test_sharepoint.py

```diff
@@ -955,6 +955,38 @@ async def test_api_call_when_token_is_expired(patch_default_wait_multiplier):
             await source.close()
 
 
+class TooManyRequestException(Exception):
+    status = 429
+    headers = {"Retry-After": 0}
+
+
+@pytest.mark.asyncio
+async def test_api_call_when_status_429_exception():
```

Comment:
If we receive a 429 response, we retry. This test doesn't retry because the retry_count limit starts at `1`.
Can you add tests that checks that we retry the expected number of times?
- 1 test for 429 errors (and if possible, test that we retry using the expected header response as the interval value).
- 1 test for non-429 errors (to make sure retrying works in general).

Reply:
I have updated test case for `test_api_call_when_status_429_exception`. And for non 429 errors I have already added test cases for that such as: `test_api_call_when_server_is_down`, `test_api_call_when_token_is_expired`.